hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/bucket/BucketCache.java

```diff
@@ -1081,8 +1101,47 @@ private void retrieveFromFile(int[] bucketSizes) throws IOException {
         throw new IOException("Persistence file does not start with protobuf magic number. " +
             persistencePath);
       }
+      byte[] pbuf2 = new byte[pblen];
```

Comment:
All these logic will go away. From the PB message we can get.. So getting back the checksum should be included as part of parsePB()

Reply:
Has been modified to use PB persistent checksum. But I have not designed how to maintain backward compatibility.